plugins/star-rating/api/api.js

```diff
@@ -10,30 +10,373 @@ var plugin = {},
-        var events = (params.qstring && params.qstring.events);
-        if (events && events.length && Array.isArray(events)) {
-            events.forEach(function (event) {
-                if (event.key === '[CLY]_star_rating') {
-                    event.segmentation['platform_version_rate'] =
-                        event.segmentation['platform'] + "**" +
-                            event.segmentation['app_version'] + "**" +
-                            event.segmentation['rating'] + "**";
+        if (params.qstring.events && params.qstring.events.length && Array.isArray(params.qstring.events)) {
+            params.qstring.events = params.qstring.events.filter(function(currEvent){
+                if (currEvent.key == "[CLY]_star_rating"){
+                    
+                    common.db.collection('apps').findOne({'key': params.qstring.app_key}, (err, app) => {
+                        if (!app) {
+                            if (plugins.getConfig("api").safe) {
+                                common.returnMessage(params, 400, 'App does not exist');
+                            }
+                            return false;
+                        }
+                        
+                        var collectionName = 'feedback' + app["_id"];
+                        
+                        common.db.collection(collectionName)
+                            .insert({
+                                "email": currEvent.segmentation.email,
+                                "comment": currEvent.segmentation.comment,
+                                "ts":currEvent.timestamp,
+                                "device_id":params.qstring.device_id,
+                                "cd":new Date(),
+                                "uid":params.app_user.uid,
+                                "contact_me":currEvent.segmentation.contactMe,
+                                "rating":currEvent.segmentation.rating,
+                                "widget_id":currEvent.segmentation.widget_id
+                            }, function(err, saved) {
+                                /**
+                                 *  register for process new  rating event data.
+                                 *  the original event format like:
+                                 *  { key: '[CLY]_star_rating', count:1, sum:1, segmentation:{ platform:"iOS", version:"3.2", rating:2}
+                                 *  this function will add a field call "platform_version_rate" in segmentation.
+                                 */
+                                currEvent.segmentation['platform_version_rate'] = currEvent.segmentation['platform'] + "**" + currEvent.segmentation['app_version'] + "**" + currEvent.segmentation['rating'] + "**";
+                                // remove data from core data
+                                // for GDPR compatibility
+                            });
+                    });
+                    return true;
```

Comment:
So you only return true if event key **[CLY]_star_rating**
Meaning all other events are filtered out and never make it to be processed :D